development/generate-icon-names.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Generate icon names
+ *
+ * Reads all the icon svg files in app/images/icons
+ * and returns an object of icon name key value pairs
+ * stored in the environment variable ICON_NAMES
+ * Used with the Icon component in ./ui/component-library/icon
+ */
+const fs = require('fs');
+const path = require('path');
+
+const SVG_ICONS_FOLDER = './app/images/icons';
+const ASSET_EXT = '.svg';
+
+const getIconNameKebabCase = (fileName) =>
+  path.basename(fileName, ASSET_EXT).replace('icon-', '');
+
+const getIconNameInSnakeCase = (fileName) =>
+  path
+    .basename(fileName, ASSET_EXT)
+    .replace('icon-', '')
+    .replace(/-/gu, '_')
+    .toUpperCase();
+
+const generateIconNames = async () => {
+  const iconNames = {};
+
+  const svgIconsFolderPath = path.join(__dirname, `../${SVG_ICONS_FOLDER}`);
+
+  const fileList = await fs.promises.readdir(svgIconsFolderPath);
+
+  const svgIconsFileList = fileList.filter(
+    (fileName) => path.extname(fileName) === ASSET_EXT,
+  );
+
+  svgIconsFileList.forEach(
+    (fileName) =>
+      (iconNames[getIconNameInSnakeCase(fileName)] =
+        getIconNameKebabCase(fileName)),
+  );
+
+  console.log('ICON_NAMES env var successfully generated!');
```

Comment:
This `console.log` might be unnecessary... Let me know @Gudahtt 

![Screen Shot 2022-09-15 at 12 52 42 PM](https://user-images.githubusercontent.com/8112138/190496680-fd749260-86ff-4f25-ba5e-bbfcf494124e.png)

Reply:
Probably would be better to omit it, since a failure to generate it properly will result in an error. Not a big deal though

Reply:
I create a PR to remove it next week